source/controllers/src/impedance/CartesianTwistController.cpp

```diff
@@ -0,0 +1,120 @@
+#include "controllers/impedance/CartesianTwistController.hpp"
+
+namespace controllers::impedance {
+
+using namespace state_representation;
+
+CartesianTwistController::CartesianTwistController(double linear_principle_damping,
+                                                   double linear_orthogonal_damping,
+                                                   double angular_stiffness,
+                                                   double angular_damping) :
+    linear_principle_damping_(std::make_shared<Parameter<double>>("linear_principle_damping",
+                                                                  linear_principle_damping)),
+    linear_orthogonal_damping_(std::make_shared<Parameter<double>>("linear_orthogonal_damping",
+                                                                   linear_orthogonal_damping)),
+    angular_stiffness_(std::make_shared<Parameter<double>>("angular_stiffness", angular_stiffness)),
+    angular_damping_(std::make_shared<Parameter<double>>("angular_damping", angular_damping)),
+    dissipative_ctrl_(ComputationalSpaceType::LINEAR),
+    velocity_impedance_ctrl_(Eigen::MatrixXd::Zero(6, 6), Eigen::MatrixXd::Zero(6, 6)) {
+  set_angular_gains(angular_stiffness, angular_damping);
+}
+
+CartesianTwistController::CartesianTwistController(const Eigen::Vector4d& gains) :
+    CartesianTwistController(gains(0), gains(1), gains(2), gains(3)) {
+
+}
+
+CartesianTwistController::CartesianTwistController(const CartesianTwistController& other) :
+    linear_principle_damping_(std::make_shared<Parameter<double>>("linear_principle_damping", other.get_gains()(0))),
+    linear_orthogonal_damping_(std::make_shared<Parameter<double>>("linear_orthogonal_damping", other.get_gains()(1))),
+    angular_stiffness_(std::make_shared<Parameter<double>>("angular_stiffness", other.get_gains()(2))),
+    angular_damping_(std::make_shared<Parameter<double>>("angular_damping", other.get_gains()(3))),
+    dissipative_ctrl_(other.dissipative_ctrl_),
+    velocity_impedance_ctrl_(other.velocity_impedance_ctrl_) {
+}
```

Comment:
what is the standard for constructors with only initializers? Should there be a line break between the `{}`? I am not sure we are consistent with that

Reply:
I think we don't have an explicit rule for it. I agree the `{}` is probably nicer, so let's go with that for now.